### ISSUE TITLE
Tuple and/or list type columns - deprecation warning

### ISSUE THREAD

**Issue by callumwebb** [astropenguin/pandas-dataclasses#125]
Hi @astropenguin, thank you very much for creating this package! I think it's excellent.

I'm interested in creating a pandas dataclass that includes a tuple or list column. I can achieve the desired result with the following code, but I do get a warning. I think I understand why the warning is raised, but it's not clear to me if and where this could be avoided from within the package.

If there's a better way to achieve the desired result please let me know (could we be more specific than `Data[object]`, e.g. `Data[tuple(int)]`?)

```python
from dataclasses import dataclass
from pandas_dataclasses import Data, AsFrame

@dataclass
class Tuples(AsFrame):

    values: Data[object]


df = Tuples.new(
    [(1, 2), (1,)]
)

df
```

`VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  return asarray(a).ndim`

**Comment by astropenguin** [comment]
Hi @callumwebb, thank you for your interest in this package! I think your code is perfect: The warning is raised when the package checks whether incoming data (i.e.  `[(1, 2), (1,)]`) is scalar or not by the `numpy.ndim` function. Let me fix the checking so as not to raise the warning in the next release.

**Comment by astropenguin** [comment]
I released v0.10.0 where I fixed the issue above. Now your code will not raise the warning!
